File: src/data/class_label/class_label_multi.py
from data.class_label.abstract_class_label import AbstractClassLabel


class ClassLabelMulti(AbstractClassLabel):
    def __init__(self, class_label):
        super().__init__(class_label)

    def __eq__(self, other):
        if not isinstance(other, ClassLabelMulti) or self.get_length() != other.get_length():
            return False

        for i in range(self.get_length()):
            if self.get_class_label_value()[i] != other.get_class_label_value()[i]:
                return False
        return True

    def get_length(self):
        return len(self.get_class_label_value())

    def __copy__(self):
        return ClassLabelMulti(self.get_class_label_value())

    def __str__(self):
        if self.get_class_label_value() is None:
            raise Exception("class label value is None")
        txt = f"{self.get_class_label_value()[0]:2d}"

        if self.get_length() > 1:
            for i in range(1, self.get_length()):
                txt = f"{txt}, {self.get_class_label_value()[i]:2d}"

        return txt

File: src/mofgbmlpy/core/data/class_label/class_label_multi.cpp
#include "class_label_multi.hpp"
#include <stdexcept>

ClassLabelMulti::ClassLabelMulti(const std::vector<int>& class_label) 
    : AbstractClassLabel(), class_label(class_label) {
}

ClassLabelMulti::ClassLabelMulti(const ClassLabelMulti& other) 
    : AbstractClassLabel(other), class_label(other.class_label) {
}


int ClassLabelMulti::get_length() const
{
    return class_label.size();
}

const std::vector<int>& ClassLabelMulti::get_class_label_value() const {
    return class_label;
}

int ClassLabelMulti::get_class_label_value_at(int index) const
{
    return class_label.at(index);
}

void ClassLabelMulti::set_class_label_vector(const std::vector<int>& class_label)
{
    this->class_label = class_label;
}

ClassLabelMulti::operator std::string() const
{
    std::string txt = std::to_string(class_label[0]);
    if (class_label.size() > 1) {
        for (size_t i = 1; i < class_label.size(); ++i) {
            txt += ", " + std::to_string(class_label[i]);
        }
    }
    return txt;
}

bool ClassLabelMulti::operator==(const AbstractClassLabel& other) const
{
    const ClassLabelMulti* other_multi = dynamic_cast<const ClassLabelMulti*>(&other);
    if (!other_multi) {
        return false;
    }
    return class_label == other_multi->class_label;
}

ClassLabelMulti* ClassLabelMulti::clone() const
{
    return new ClassLabelMulti(*this);
}

void ClassLabelMulti::set_class_label_value(const std::vector<int>& new_value) {
    class_label = new_value;
}

